modules/xfem/include/efa/EFAFuncs.h

```diff
@@ -0,0 +1,83 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef EFAFUNCS_H
+#define EFAFUNCS_H
+
+#include <map>
+#include <set>
+#include <vector>
+#include <algorithm>
+
+template <typename T>
+bool deleteFromMap(std::map<unsigned int, T*> &theMap, T* elemToDelete)
+{
+  bool didIt = false;
+  typename std::map<unsigned int, T*>::iterator i=theMap.find(elemToDelete->id());
+  if (i != theMap.end())
+  {
+    delete i->second;
+    theMap.erase(i);
```

Comment:
As the code is structured, it would be trivial to pull this algorithm out and use it in another code, so I don't want to rule that possibility out. We need to talk offline about what the options for smart pointers would be.